core/drivers/fabric-driver/server/events.ts

```diff
@@ -0,0 +1,323 @@
+        await db?.close();
+        throw new Error(error);
+    }
+}
+
+const deleteEventSubscription = async (
+    eventMatcher: eventsPb.EventMatcher,
+    requestId: string
+): Promise<eventsPb.EventSubscription> => {
+    console.log(`deleting from driver subscription of the eventMatcher: ${JSON.stringify(eventMatcher.toObject())} with requestId: ${requestId}`);
+    var subscriptions: Array<string>;
+    var retVal: eventsPb.EventSubscription = new eventsPb.EventSubscription();
+    retVal.setEventMatcher(eventMatcher);
+    let db: DBConnector;
+    try {
+        // Create connection to a database
+        db = new LevelDBConnector(DB_NAME!);
+        await db.open();
+
+        const key: string = Buffer.from(eventMatcher.serializeBinary()).toString('base64');
+        try {
+            // fetch the current values in the DB against the given key
+            var subscriptionsSerialized: string = await db.read(key) as string;
+            subscriptions = JSON.parse(subscriptionsSerialized);
+
+            console.debug(`subscriptions.length: ${subscriptions.length}`);
+            var foundEntry: boolean = false;
+            for (var subscriptionSerialized of subscriptions) {
+                var subscription: queryPb.Query =  queryPb.Query.deserializeBinary(Buffer.from(subscriptionSerialized, 'base64'));
+                if (subscription.getRequestId() == requestId) {
+                    console.debug(`deleting the subscription (with input requestId): ${JSON.stringify(subscription.toObject())}`);
+                    subscriptions.splice(subscriptions.indexOf(subscriptionSerialized), 1);
+                    retVal.setQuery(subscription);
+                    foundEntry = true;
+                    break;
+                }
+            }
+
+            if (!foundEntry) {
+                throw new Error(`event subscription with requestId: ${requestId} is not found!`);
+            }
+        } catch (error: any) {
+            // error could be either due to key not being present in the database or some other issue with database access
+            console.error(`re-throwing error: ${JSON.stringify(error)}`);
+            await db.close();
+            throw new Error(error);
+        }
+
+        console.debug(`subscriptions.length: ${subscriptions.length}`);
+        if (subscriptions.length == 0) {
+            await db.delete(key);
+        } else {
+            subscriptionsSerialized = JSON.stringify(subscriptions);
+            await db.insert(key, subscriptionsSerialized);
+        }
+
+        await db.close();
+        console.log(`end deleteEventSubscription() .. retVal: ${JSON.stringify(retVal.toObject())}`);
+        return retVal;
+    } catch(error: any) {
+        console.error(`Error during delete: ${JSON.stringify(error)}`);
+        await db?.close();
+        throw new Error(error);
+    }
+}
+
+async function lookupEventSubscriptions(
+    eventMatcher: eventsPb.EventMatcher
```

Comment:
Add a function that does exactly what this function does but takes as parameters: <channel-id, contract-id, func-name>.

You can extract most of the code from here into that function and just call that from here by extracting the above values from the `eventMatcher` parameter.

Reply:
Done (updated the above function itself so that matching based on EventMatcher properties can happen)!